### PR TITLE
Use shared CI template

### PR DESCRIPTION
**Pull request by daniel-m-dd**
### What does this PR do?

Use a shared template for some CI jobs.

### Motivation

What inspired you to submit this pull request?

### Additional Notes

Anything else we should know when reviewing?

### Review checklist (to be filled by reviewers)

- [ ] Feature or bugfix MUST have appropriate tests (unit, integration, e2e)
- [ ] Make sure you discussed the feature or bugfix with the maintaining team in an Issue
- [ ] Make sure each commit and the PR mention the Issue number (cf the [CONTRIBUTING](CONTRIBUTING.md) doc)

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -34,3 +33,5 @@ allprojects {
 task<Delete>("clean") {
     delete(rootProject.buildDir)
 }
+
+tasks.register("checkGeneratedFiles") {}
```
Comment (0xnm) [inline build.gradle.kts:37]: seems it is some leftover?

File: samples/basic/src/main/java/com/example/variants/MainActivity.kt [hunk 2]
```diff
@@ -3,7 +3,6 @@ package com.example.variants
 import android.os.Bundle
 import androidx.appcompat.app.AppCompatActivity
 // unused import is on purpose
-import com.example.lib.Placeholder
```
Comment (0xnm) [inline samples/basic/src/main/java/com/example/variants/MainActivity.kt:6]: we shouldn't remove this line

File: samples/variants/src/main/java/com/example/variants/MainActivity.kt [hunk 3]
```diff
@@ -1,9 +1,8 @@
 package com.example.variants
 
+// unused import is on purpose
 import android.os.Bundle
 import androidx.appcompat.app.AppCompatActivity
-// unused import is on purpose
-import com.example.lib.Placeholder
```
Comment (0xnm) [inline samples/variants/src/main/java/com/example/variants/MainActivity.kt:6]: same - shouldn't remove this line

**Review by 0xnm — CHANGES_REQUESTED** [review]

File: build.gradle.kts (hunk 1 above)
Comment (daniel-m-dd) [reply build.gradle.kts:37]: it is actually needed by the shared pipeline (but no use here so it is a task doing nothing)

Comment (0xnm) [reply build.gradle.kts:37]: can we then add a comment? so that empty task is not confusing

**Review by 0xnm — APPROVED** [review]

**Review by xgouchet — APPROVED** [review]

**Pull request merged**